File: hw2/java/Order.java
public class Order {

    private boolean refunded = false;
    private Meal meal;
    private Philosopher orderedBy;

    public Order(int seatNo, Meal meal, Philosopher orderedBy) {
        this.meal = meal;
        this.orderedBy = orderedBy;
    }

    public double price() {
        return meal.price();
    }

    public void markAsRefund() {
        this.refunded = true;
    }

    public boolean isRefunded() {
        return this.refunded;
    }

    public Meal meal() {
        return this.meal;
    }

    public Philosopher orderedBy() {
        return this.orderedBy;
    }
}

File: hw2/go/order.go
package main

type Order struct {
	orderedBy *Philosopher
	refunded  bool
	meal      Meal
}

func (o *Order) markRefunded() {
	o.refunded = true
}

func (o *Order) isRefunded() bool {
	return o.refunded
}

func (o *Order) price() float64 {
	return o.meal.price
}

func (o *Order) name() string {
	return o.meal.getName()
}
